examples/containers/multi/helpers/driver.sh

```diff
@@ -0,0 +1,222 @@
+# - The passed docker container name/ID is valid and running
+# - The command passed is a valid bash command to be executed inside the container
+# - $success_msg is returned by the command upon successful execution
+
+# Post-conditions:
+# - The command is executed successfully with no error code; OR
+# - The command fails $retry_count times in a row and the program exits
+function _docker_exec_with_retry() {
+
+    local retry_count=5
+    local retry_time=(2 4 8 15 30)
+
+    local container=$1
+    local command=$2
+    local success_msg=${3:-""}
+
+    local i=0
+    for ((i; i<retry_count; i++)); do
+        output=$(docker exec "$container" /usr/bin/env bash -c "$command")
+        if [[ $output == *$success_msg* ]]; then
+            return
+        else
+            echo "Command failed, waiting ${retry_time[$i]} seconds before retrying..."
+            sleep "${retry_time[$i]}"
+            
+        fi
+    done
+    echo "Max retries reached while executing $command, $container failed for reason: $output"
+    exit 1
+
+}
+# Pre-conditions: 
+#   - $num_nodes containers running Couchbase Server (uninitialised)/exporter 
+
+# Post-conditions: 
+#   - All CBS/exporter nodes initialised and partitioned as evenly as possible into 
+#     $num_clusters clusters, with a rebalance occurring after the last node is added
+#   - $num_clusters nodes will be running the Data Service, the rest Index/Query, with quotas
+#     specified by $data_alloc and $index_alloc
+#   - Every cluster registered for monitoring with the cbmultimanager
+function configure_servers() {
+
+    local num_nodes=$1
+    local num_clusters=$2
+    local server_user=$3
+    local server_pwd=$4
+    local node_ram=$5
+    local load=$6
+
+    local data_alloc 
+    local index_alloc
+    # Allocate 70% of the specified RAM quota to the service (query has no quota)
+    # awk used as bash does not support operations with decimals
+    data_alloc=$(awk -v n="$node_ram" 'BEGIN {printf "%.0f\n", (n*0.7)}')
+    index_alloc=$(awk -v n="$node_ram" 'BEGIN {printf "%.0f\n", (n*0.7)}')
+
+    local sample_buckets=(\"travel-sample\" \"beer-sample\") # Only used if LOAD is true
+
+    local temp_dir
+    temp_dir=$(mktemp -d)
+    echo '[]' > "$temp_dir"/targets.json
+
+    echo "----- START CONFIGURING NODES -----"
+    echo "Partitioning $num_nodes nodes into $num_clusters clusters..."
+    echo ""
+
+    local nodes_left=$num_nodes
+    local i=0
+    for ((i; i<num_clusters; i++)); do
+
+        # Calculate the number of nodes to provision in this cluster
+        local to_provision=$(( nodes_left / (num_clusters - i) )) # (Integer division, Bash does not support decimals)
+        local start=$(( num_nodes - nodes_left ))
+        
+        # Create and initialize cluster
+        local uid="node$start"
+        local clust_name="Cluster $i"
+        _docker_exec_with_retry "$uid" "/opt/couchbase/bin/couchbase-cli cluster-init -c localhost --cluster-name=\"$clust_name\" \
```

Comment:
NITPICK: This might be easier if `_docker_exec_with_retry` simply forwarded its arguments onto `docker exec`, which would mean you can dispense with quoting everything.